src/mongocxx/CMakeLists.txt

```diff
@@ -135,78 +135,71 @@ include_directories(
     ${CMAKE_BINARY_DIR}/src
 )
 
-# We define two libraries: the normal library and the testing-only library.  The testing-only
-# library does not get installed, but the tests link against it instead of the normal library.  The
-# only difference between the two libraries is that MONGOCXX_TESTING is defined in the testing-only
-# library (this define enables special testing-only behavior).
-add_library(mongocxx
-    ${mongocxx_sources}
-)
-add_library(mongocxx_mocked
-    ${mongocxx_sources}
-)
-target_compile_definitions(mongocxx_mocked PUBLIC MONGOCXX_TESTING)
-
-if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    target_compile_options(mongocxx_mocked PRIVATE /bigobj)
-endif()
-
-set_target_properties(mongocxx_mocked PROPERTIES
-    OUTPUT_NAME mongocxx-mocked
-)
+function(mongocxx_add_library TARGET OUTPUT_NAME LINK_TYPE)
+    add_library(${TARGET} ${LINK_TYPE}
+        ${mongocxx_sources}
+    )
 
-if (BUILD_SHARED_LIBS)
-    set_target_properties(mongocxx PROPERTIES
-        OUTPUT_NAME mongocxx
+    set_target_properties (${TARGET} PROPERTIES
+        OUTPUT_NAME ${OUTPUT_NAME}
+        VERSION ${MONGOCXX_VERSION}
+        DEFINE_SYMBOL MONGOCXX_EXPORTS
     )
 
-    foreach (TARGET mongocxx mongocxx_mocked)
+    if(LINK_TYPE STREQUAL "SHARED")
         set_target_properties (${TARGET} PROPERTIES
-            VERSION ${MONGOCXX_VERSION}
-            DEFINE_SYMBOL MONGOCXX_EXPORTS
             CXX_VISIBILITY_PRESET hidden
             VISIBILITY_INLINES_HIDDEN ON
             SOVERSION ${MONGOCXX_ABI_VERSION}
         )
-    endforeach(TARGET)
-else()
-    target_compile_definitions(mongocxx PUBLIC MONGOCXX_STATIC)
-    target_compile_definitions(mongocxx_mocked PUBLIC MONGOCXX_STATIC)
+    endif()
 
-    set_target_properties(mongocxx PROPERTIES
-        OUTPUT_NAME mongocxx-static
-    )
-endif()
+    if(LINK_TYPE STREQUAL "STATIC")
+        # Nothing
```

Comment:
Do we need `target_compile_definitions(${TARGET} PUBLIC MONGOCXX_STATIC)`?

Reply:
So we don't actually use that symbol internally as far as I can tell, we're handing that to other projects that include our headers via pkg-config or cmake config. That said, it really doesn't hurt to do it. (I think I missed it when I did my original conversion.)